src/react/RegisterDevicePage.tsx

```diff
@@ -34,48 +42,42 @@ import { useAlerts } from './AlertManager';
  */
 function pasteSecret() {
   const secretCode = document.querySelector('#secret-code');
+  if (secretCode == null) {
+    return;
+  }
   const selection = window.getSelection();
-
+  if (selection == null) {
+    return;
+  }
   if (selection.rangeCount > 0) {
     selection.removeAllRanges();
   }
-
   const range = document.createRange();
   range.selectNode(secretCode);
   selection.addRange(range);
   document.execCommand('copy');
 }
 
-const ColNoLabel = ({ sm, className, ...otherProps }) => (
+type ColNoLabelProps = React.ComponentProps<typeof Col>;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const ColNoLabel = ({ sm, className = '', ...otherProps }: ColNoLabelProps): React.ReactElement => (
   <Col sm="auto" className={'col-no-label '.concat(className)} {...otherProps} />
 );
 
-const InstrospectionTable = ({ interfaces, onAddInterface, onRemoveInterface }) => (
-  <Table className="mb-4" responsive>
-    <thead>
-      <tr>
-        <th>Interface name</th>
-        <th>Major</th>
-        <th>Minor</th>
-        <th className="action-column"> </th>
-      </tr>
-    </thead>
-    <tbody>
-      {Array.from(interfaces).map(([key, interfaceDescriptor]) => (
-        <InterfaceIntrospectionRow
-          key={key}
-          name={interfaceDescriptor.name}
-          major={interfaceDescriptor.major}
-          minor={interfaceDescriptor.minor}
-          onRemove={() => onRemoveInterface(interfaceDescriptor)}
-        />
-      ))}
-      <IntrospectionControlRow onAddInterface={onAddInterface} />
-    </tbody>
-  </Table>
-);
+interface InterfaceIntrospectionRowProps {
```

Comment:
this looks pretty similar to the `InterfaceDescriptor` type. Maybe accept a `InterfaceDescriptor` and a onRemove callback?

Reply:
That's right, fixed it.